enterprise_access/apps/api/v1/views.py

```diff
@@ -746,7 +746,128 @@ def partial_update(self, request, *args, **kwargs):
+    filter_backends = (filters.OrderingFilter, DjangoFilterBackend,)
+    filterset_fields = ('group_uuid', 'policy_type',)
+    pagination_class = PaginationWithPageCount
+    http_method_names = ['get', 'head', 'options', 'post', 'patch', 'delete']
+    lookup_field = 'uuid'
+    lookup_value_regex = '[0-9a-f-]+'
+    permission_required = 'requests.has_admin_access'
+
+    @property
+    def requested_group_uuid(self):
+        """
+        The group uuid from request params or post body
+        """
+        group_uuid = None
+        if self.action in ('retrieve', 'partial_update', 'destroy'):
+            policy_uuid = self.kwargs.get('uuid')
+            if policy_uuid:
+                try:
+                    policy = SubsidyAccessPolicy.objects.get(uuid=policy_uuid)
+                    group_uuid = str(policy.group_uuid)
+                except ObjectDoesNotExist:
+                    group_uuid = None
+        elif self.action == 'create':
+            group_uuid = self.request.POST.get('group_uuid')
+        else:
+            group_uuid = self.request.query_params.get('group_uuid', None)
+        return group_uuid
+
+    def get_permission_object(self):
+        """
+        Returns the enterprise uuid (group uuid) to verify that requesting user possess the enterprise admin role.
+        """
+        return self.requested_group_uuid
+
+    def create(self, request, *args, **kwargs):
+        """
+        create action for SubsidyAccessPolicyCRUDViewset. Handles creation of policy after validation
+        """
+        policy_data = request.data
+        serializer = self.get_serializer(data=policy_data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def _validate_policy(self, policy_uuid=None, policy_data=None):
+        """
+        Args:
+            policy_uuid: a list of one or more valid uuid strings
+        Raises:
+            SubsidyAccessPolicyError: if policy uuid is invalid or there is an attempt to change policy uuid.
+        Returns:
+            policy instance
+        """
+        if policy_uuid:
+            try:
+                instance = SubsidyAccessPolicy.objects.get(uuid=policy_uuid)
+            except ObjectDoesNotExist as exc:
+                error_msg = f'Policy UUID provided ({policy_uuid}) is not a valid UUID'
+                logger.exception(error_msg)
+                raise SubsidyAccessPolicyError(error_msg, status.HTTP_400_BAD_REQUEST) from exc
+            # for updates, ensure that uuid is not editable
+            if instance and policy_data.get('uuid') != str(instance.uuid):
+                error_msg = f'Policy UUID cannot be changed'
+                logger.exception(error_msg)
+                raise SubsidyAccessPolicyError(error_msg, status.HTTP_400_BAD_REQUEST)
+            return instance
```

Comment:
I think this can be moved into serializer.

On a second thought, `uuid` of the policy will be auto-generated. So I think we should not consider uuid coming from client. In that case I think we don't need to do any validation for uuid for put/post/patch?

Reply:
For POST/PUT, yes. For PATCH, the interface is more flexible in customizing a payload. So admin can easily add a key-value for uuid in the payload. Just being extra cautious here. I had initially added this logic in serializer but obviously, it automatically removes uuid from payload so this condition `policy_data.get('uuid') != str(instance.uuid)` always passes as left side is always None.